applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/adjoint_nonlinear_strain_energy_response_function.cpp

```diff
@@ -0,0 +1,378 @@
+        {
+            if( response_gradient_0[i] != 0 )
+            {
+                rResponseGradient[0] = response_gradient_1[i] / response_gradient_0[i];
+                break;
+            }
+        }
+
+        mResponseGradient_0[rAdjointCondition.Id()] = response_gradient_1;
+        KRATOS_CATCH("");
+    }
+
+    void AdjointNonlinearStrainEnergyResponseFunction::CalculatePartialSensitivity(Element& rAdjointElement,
+                                                const Variable<array_1d<double, 3>>& rVariable,
+                                                const Matrix& rSensitivityMatrix,
+                                                Vector& rSensitivityGradient,
+                                                const ProcessInfo& rProcessInfo)
+    {
+        KRATOS_TRY
+
+        if (rSensitivityGradient.size() != 0)
+            rSensitivityGradient.resize(0, false);
+
+        KRATOS_CATCH("");
+    }
+
+
+    void AdjointNonlinearStrainEnergyResponseFunction::CalculatePartialSensitivity(Element& rAdjointElement,
+                                                const Variable<double>& rVariable,
+                                                const Matrix& rSensitivityMatrix,
+                                                Vector& rSensitivityGradient,
+                                                const ProcessInfo& rProcessInfo)
+    {
+        KRATOS_TRY;
+
+        if (rSensitivityGradient.size() != 0)
+            rSensitivityGradient.resize(0, false);
+
+        KRATOS_CATCH("");
+    }
+
+    void AdjointNonlinearStrainEnergyResponseFunction::CalculatePartialSensitivity(Condition& rAdjointCondition,
+                                                const Variable<double>& rVariable,
+                                                const Matrix& rSensitivityMatrix,
+                                                Vector& rSensitivityGradient,
+                                                const ProcessInfo& rProcessInfo)
+    {
+        KRATOS_TRY;   
+
+        if (rSensitivityGradient.size() != 0)
+            rSensitivityGradient.resize(0, false);
+        
+        
+        
+        KRATOS_CATCH("");
+    }
+
+
+    // Calculates the derivate of the response function with respect to the design parameters
+    // Computation of \frac{1}{2} (u^T_i - u^T_i-1)  \cdot ( \frac{\partial f_{ext}_i}{\partial x} + frac{\partial f_{ext}_i-1}{\partial x})
+    void AdjointNonlinearStrainEnergyResponseFunction::CalculatePartialSensitivity(Condition& rAdjointCondition,
+                                                const Variable<array_1d<double, 3>>& rVariable,
+                                                const Matrix& rSensitivityMatrix,
+                                                Vector& rSensitivityGradient,
+                                                const ProcessInfo& rProcessInfo)
+    {
+        KRATOS_TRY;
+        ProcessInfo process_info = rProcessInfo;
```

Comment:
make a todo that this should be removed since expensive

should bot be necessary any more after #2993